addons/utm/models/utm_medium.py

```diff
@@ -11,6 +11,15 @@ class UtmMedium(models.Model):
     _description = 'UTM Medium'
     _order = 'name'
 
+    REQUIRED_UTM_MEDIUMS_REF = {
```

Comment:
'mass_mailing_sms.utm_medium_sms' should probably be included, so that people cannot unlink it. Either we hardcode it here (a bit meh but less code; however it does not ease addition of medium to protect as we would have to add it here), either we do an override mechanism (a bit like "SELF_READABLE_FIELDS" on users). More code but more modular approach ... maybe check a bit what the code would look like with clean override, just to check ?

Thanks ! Sorry to annoy you, just that way we do things cleanly and take into account other UTMs in sub modules :) (by the way, don't we have some of them in social ?)

Reply:
Don't worry about, it I made a version that follows the same logic as `SELF_READABLE_FIELDS` and there's not much more code as if I would've hardcoded it in my opinion.

Also checked for other `utm.medium` created in other modules (especially social) and the only ones I see are dynamically created [here](https://github.com/odoo/enterprise/blob/b543e99a60539257e66437c0ea9fe6dee6a37bf6/social/models/social_account.py#L113-L119).
And from what I understand, they don't have any refs so there is no risk for them.